multy_test/serialized_keys_test_base.cpp

```diff
@@ -29,59 +29,68 @@ void PrintTo(const SerializedKeyTestCase& c, std::ostream* out)
          << "}";
 }
 
-TEST_P(SerializedKeyTestP, private_key_to_address)
+void SerializedKeyTestP::SetUp()
 {
-    const auto& param = GetParam();
-    const Currency currency = ::testing::get<0>(param);
-    const SerializedKeyTestCase& test_data = ::testing::get<1>(param);
-    AccountPtr account;
-    ErrorPtr error;
-    error.reset(
+    const BlockchainType blockchain_type = ::testing::get<0>(GetParam());
+    const SerializedKeyTestCase& test_data = ::testing::get<1>(GetParam());
+
+    HANDLE_ERROR(
             make_account(
-                    currency, test_data.private_key, reset_sp(account)));
-    EXPECT_EQ(nullptr, error);
+                    blockchain_type.blockchain,
+                    test_data.private_key, reset_sp(account)));
     ASSERT_NE(nullptr, account);
+}
+
+TEST_P(SerializedKeyTestP, public_key)
+{
+    const SerializedKeyTestCase& test_data = ::testing::get<1>(GetParam());
```

Comment:
add field test_data in class  SerializedKeyTestP and delete test_data here and beyond too because it you can set in void SerializedKeyTestP::SetUp()

Reply:
Doesn't worth the effort, since only few tests actually use same parts of `GetParam()`